live-build/misc/upgrade-scripts/verify-container

```diff
@@ -0,0 +1,274 @@
+	# service, such that it will pass the "--capability=all" option
+	# to "systemd-nspawn".
+	#
+	mkdir -p "/etc/systemd/system/systemd-nspawn@$CONTAINER.service.d" ||
+		die "failed to create container service override directory"
+	cat >"/etc/systemd/system/systemd-nspawn@$CONTAINER.service.d/override.conf" <<-EOF ||
+		[Service]
+		ExecStart=
+		ExecStart=/usr/bin/systemd-nspawn --quiet --boot --capability=all --machine=%i
+	EOF
+		die "failed to create container service override file"
+
+	echo "$CONTAINER"
+}
+
+function start() {
+	if [[ ! -d "/var/lib/machines/$CONTAINER" ]] ||
+		[[ ! -f "/etc/systemd/nspawn/$CONTAINER.nspawn" ]] ||
+		[[ ! -d "/etc/systemd/system/systemd-nspawn@$CONTAINER.service.d" ]] ||
+		! zfs list "rpool/ROOT/$CONTAINER" &>/dev/null; then
+		die "container '$CONTAINER' non-existent or mis-configured"
+	fi
+
+	systemctl start "systemd-nspawn@$CONTAINER" ||
+		die "failed to start container '$CONTAINER'"
+
+	#
+	# When starting the container above with "systemctl start", the
+	# command will return prior to the container having fully
+	# completed it's boot process. Thus, to ensure we don't attempt
+	# the verification steps prior to the container being ready, we
+	# first wait for the container to finish booting.
+	#
+	# shellcheck disable=SC2034
+	for i in {1..600}; do
+		if systemd-run --machine="$CONTAINER" --quiet --wait -- \
+			/bin/systemctl is-active default.target 2>/dev/null; then
+			break
+		fi
+
+		sleep 1
+	done
+
+	systemd-run --machine="$CONTAINER" --quiet --wait -- \
+		/bin/systemctl is-active default.target ||
+		die "'default.target' inactive in container '$CONTAINER'"
+}
+
+function stop() {
+	systemctl stop "systemd-nspawn@$CONTAINER"
+}
+
+function destroy() {
+	stop "$CONTAINER"
+
+	rm -f "/etc/systemd/nspawn/$CONTAINER.nspawn"
+	rm -rf "/etc/systemd/system/systemd-nspawn@$CONTAINER.service.d"
+
+	if zfs list "rpool/ROOT/$CONTAINER" &>/dev/null; then
+		zfs destroy "rpool/ROOT/$CONTAINER"
+	fi
+
+	if zfs list "rpool/ROOT/ubuntu@$CONTAINER" &>/dev/null; then
+		zfs destroy "rpool/ROOT/ubuntu@$CONTAINER"
+	fi
+
+	[[ -d "/var/lib/machines/$CONTAINER" ]] &&
+		rm -d "/var/lib/machines/$CONTAINER"
+}
+
+function usage() {
+	echo "$(basename "$0"): $*" >&2
+	echo "Usage: $(basename "$0") create"
+	echo "                        start   <container>"
+	echo "                        stop    <container>"
+	echo "                        destroy <container>"
+	exit 2
+}
+
+[[ "$EUID" -ne 0 ]] && die "must be run as root"
+[[ $# -gt 2 ]] && usage "too many arguments specified"
+[[ $# -le 0 ]] && usage "too few arguments specified"
+
+case "$1" in
+create)
+	[[ $# -gt 1 ]] && usage "too many arguments specified"
+	create
+	;;
+start)
+	[[ $# -le 1 ]] && usage "too few arguments specified"
+	CONTAINER="$2"
+	start
+	;;
+stop)
+	[[ $# -le 1 ]] && usage "too few arguments specified"
+	CONTAINER="$2"
+	stop
+	;;
+destroy)
+	[[ $# -le 1 ]] && usage "too few arguments specified"
+	CONTAINER="$2"
+	destroy
```

Comment:
It might be good to have `verify-container destroy <container>` return a non-zero exit code if the destroy actually fails.

Reply:
I'll look into this.